### DISCUSSION TITLE
When does the JointGrid space parameter come into play？

### DISCUSSION THREAD

**Discussion by flpan** [mwaskom/seaborn#3160] [Q&A]
my code:
```python
g = sns.JointGrid(x='True', y='Pred', data=mydata, space=0,
                    xlim=(-2, 23), ylim=(-2, 23),
                    height=6,
                    ratio=5,
                    )
g = g.plot_joint(sns.regplot, color = '#4F6B42')
plt.grid(linestyle='--')
g.plot_marginals(sns.distplot, kde=True,  color="#4F6B42", bins=45, hist_kws={'edgecolor':'k'})
```
 
I've set space=0, but the interval still exists.

**Comment by flpan** [comment]
mydata:
```
TRUE	Pred
8.08263 8.055517
3.369413	3.4089353
5.504575	5.2975693
7.986776	7.8860564
5.840293	5.69897
```

**Comment by mwaskom** [comment]
What “interval”? Please add an image of the result you’re getting.

**Comment by flpan** [comment]
It‘s my result:
![scatter_curve](https://user-images.githubusercontent.com/61832773/204177154-057b486d-12c2-4409-aad7-0f6102a622c9.jpg)

**Comment by flpan** [comment]
I didn't want a gap between the distribution and the scatter plot, so I set 'space=0', but it didn't work

**Reply by flpan** [reply]
My searborn version is 0.9.0, and I have tried 0.10.0 and 0.11.0

**Comment by mwaskom** [comment]
Please make your example reproducible (something I could copy and paste and run to check).

**Reply by flpan** [reply]
```python
mydata = pd.read_csv('D:/桌面/test.csv')
g = sns.JointGrid(x='True', y='Pred', data=mydata, space=0.9,
                    xlim=(-2, 25), ylim=(-2, 25),
                    height=6,
                    ratio=5,
                    )
g.ax_joint.xaxis.set_major_locator(ticker.MultipleLocator(2.5))
g.ax_joint.yaxis.set_major_locator(ticker.MultipleLocator(2.5))
g = g.plot_joint(sns.regplot, color = '#4F6B42') 
plt.grid(linestyle='--')
g.plot_marginals(sns.distplot, kde=True,  color="#4F6B42", bins=45, hist_kws={'edgecolor':'k'})
plt.show()
```
This is code

**Reply by flpan** [reply]
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.ticker as ticker

**Reply by mwaskom** [reply]
I don’t have your file so I cannot copy/paste this.

**Reply by mwaskom** [reply]
I also don’t need your specific data. Generating random x/y values should be fine.

**Reply by flpan** [reply]
> I don’t have your file so I cannot copy/paste this.

[test.csv](https://github.com/mwaskom/seaborn/files/10103775/test.csv)

**Reply by mwaskom** [reply]
> I also don’t need your specific data. Generating random x/y values should be fine.

Please don't make it harder than necessary to help you.

**Reply by mwaskom** [reply]
I can also tell that you are not making the image you showed me with this code because the y axis label would be "Pred" not "Prediction".

Please read this blog post on how to write a useful minimal example https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

**Reply by flpan** [reply]
I also used random values, but space=0 still doesn't work.
```python
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib.ticker as ticker
rs = np.random.RandomState(2)
mydata = pd.DataFrame(rs.randn(200,2), columns = ['A','B'])
g = sns.JointGrid(x=mydata['A'], y=mydata['B'], data=mydata, space=0.9,
                    xlim=(-5, 5), ylim=(-5, 5),
                    height=6,
                    ratio=5,
                    )
g.set_axis_labels('True', 'Prediction', fontsize=10.5)
g = g.plot_joint(sns.regplot, color = '#4F6B42')  # 绘制散点图
plt.grid(linestyle='--') # 设置网格线
g.plot_marginals(sns.distplot, kde=True,  color="#4F6B42", bins=45, hist_kws={'edgecolor':'k'})
plt.show()
```

**Reply by flpan** [reply]
![1669639854109](https://user-images.githubusercontent.com/61832773/204282229-e4e5d49a-7c8d-49ea-b7a0-84ed1c2b85ca.jpg)
this is the result!

**Reply by flpan** [reply]
> I can also tell that you are not making the image you showed me with this code because the y axis label would be "Pred" not "Prediction".
> 
> Please read this blog post on how to write a useful minimal example https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports

I use `g.set_axis_labels('True', 'Prediction', fontsize=10.5)` to modify the label to “Prediction”.

**Reply by flpan** [reply]
```python
rs = np.random.RandomState(2)
#产生200行2列个0-2之间的值
mydata = pd.DataFrame(rs.randn(200,2), columns = ['A','B'])

g = sns.JointGrid(x=mydata['A'], y=mydata['B'], data=mydata, space=0.9,
                    xlim=(-5, 5), ylim=(-5, 5),
                    height=6,
                    ratio=5,
                    )
g = g.plot_joint(sns.regplot)
g.plot_marginals(sns.distplot, color="#4F6B42", hist_kws={'edgecolor':'k'})
plt.show()
```
this is a simple code.

**Reply by flpan** [reply]
The result:
![1669640944677](https://user-images.githubusercontent.com/61832773/204285690-59754648-68e8-4ace-8b5a-6dbb732d8617.jpg)

**Reply by mwaskom** [reply]
You have `space=0.9` in that script. When I change it to `space=0` it works as expected.

**Reply by flpan** [reply]
> You have `space=0.9` in that script. When I change it to `space=0` it works as expected.

I'm sorry, I have make the `space=0`, it don't work as expected. I wonder if this is an environmental problem.
numpy                     1.23.4
matplotlib                3.5.3
pandas                    1.5.1
seaborn                   0.9.0
python                    3.8.5

**Reply by flpan** [reply]
> You have `space=0.9` in that script. When I change it to `space=0` it works as expected.

What is your environment configuration like?

**Reply by flpan** [reply]
> You have `space=0.9` in that script. When I change it to `space=0` it works as expected.

Hello, could you please show me the environment configuration? I really can't solve it. Also, I have changed the 'space' in 'class JointGrid(object):', but it still doesn't work.

**Comment by flpan** [comment]


**Reply by flpan** [reply]
Thank you for your reply
